extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/devmode/KafkaStreamsHotReplacementSetup.java

```diff
@@ -32,11 +33,14 @@ public void run() {
                             @Override
                             public void run() {
                                 try {
+                                    KafkaStreamsProducer.ignoreShutdownFlag = true;
```

Comment:
Won't this cause things to not shut down properly? If a shutdown request comes in while this flag is set, this request will be ignored, but it seems we'd never do the shutdown later on then?

Reply:
I think @rquinio's [comment](https://github.com/quarkusio/quarkus/pull/12224#discussion_r492005054) make sense in fact.
The producer bean is recreated however, the class is still the same as loaded before, however shutdown flag is set to true because if the first shutdown from dev mode. So putting the state back to the initial value in the constructor should make the trick